File: japiwdbe/WdbeQTrgList.java
/**
	* \file WdbeQTrgList.h
	* Java API code for record of table TblWdbeQTrgList
	* \copyright (C) 2018-2020 MPSI Technologies GmbH
	* \author Alexander Wirthmueller (auto-generation)
	* \date created: 5 Dec 2020
  */
// IP header --- ABOVE

package apiwdbe;

import org.w3c.dom.*;
import sbecore.*;

public class WdbeQTrgList {
	
	public WdbeQTrgList(
				int jnum
				, String sref
				, String stubSysRefWdbeMSystem
				, String stubRefWdbeMUnit
			) {
		this.jnum = jnum;
		this.sref = sref;
		this.stubSysRefWdbeMSystem = stubSysRefWdbeMSystem;
		this.stubRefWdbeMUnit = stubRefWdbeMUnit;
	};

	public int jnum;
	public String sref;
	public String stubSysRefWdbeMSystem;
	public String stubRefWdbeMUnit;
	
	public boolean readXML(
				Document doc
				, String basexpath
				, boolean addbasetag
			) {
		if (addbasetag) basexpath = Xmlio.checkUclcXPaths(doc, basexpath, "WdbeQTrgList");

		if (Xmlio.checkXPath(doc, basexpath)) {
			sref = Xmlio.extractStringUclc(doc, basexpath, "sref", "srf", null, 0);
			stubSysRefWdbeMSystem = Xmlio.extractStringUclc(doc, basexpath, "stubSysRefWdbeMSystem", "sys", null, 0);
			stubRefWdbeMUnit = Xmlio.extractStringUclc(doc, basexpath, "stubRefWdbeMUnit", "unt", null, 0);
			
			return true;
		};

		return false;
	};
};

File: apiwdbe/WdbeQTrgList.cpp
/**
	* \file WdbeQTrgList.cpp
	* API code for table TblWdbeQTrgList (implementation)
	* \author Alexander Wirthmueller
	* \date created: 11 Jul 2020
	* \date modified: 11 Jul 2020
	*/

#include "WdbeQTrgList.h"

using namespace std;
using namespace Sbecore;
using namespace Xmlio;

/******************************************************************************
 class WdbeQTrgList
 ******************************************************************************/

WdbeQTrgList::WdbeQTrgList(
			const uint jnum
			, const string sref
			, const string stubSysRefWdbeMSystem
			, const string stubRefWdbeMUnit
		) {
	this->jnum = jnum;
	this->sref = sref;
	this->stubSysRefWdbeMSystem = stubSysRefWdbeMSystem;
	this->stubRefWdbeMUnit = stubRefWdbeMUnit;
};

bool WdbeQTrgList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "WdbeQTrgList");
	else
		basefound = checkXPath(docctx, basexpath);

	if (basefound) {
		extractStringUclc(docctx, basexpath, "sref", "srf", sref);
		extractStringUclc(docctx, basexpath, "stubSysRefWdbeMSystem", "sys", stubSysRefWdbeMSystem);
		extractStringUclc(docctx, basexpath, "stubRefWdbeMUnit", "unt", stubRefWdbeMUnit);
	};

	return basefound;
};

/******************************************************************************
 class ListWdbeQTrgList
 ******************************************************************************/

ListWdbeQTrgList::ListWdbeQTrgList() {
};

ListWdbeQTrgList::ListWdbeQTrgList(
			const ListWdbeQTrgList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WdbeQTrgList(*(src.nodes[i]));
};

ListWdbeQTrgList& ListWdbeQTrgList::operator=(
			const ListWdbeQTrgList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WdbeQTrgList(*(src.nodes[i]));

	return *this;
};

ListWdbeQTrgList::~ListWdbeQTrgList() {
	clear();
};

void ListWdbeQTrgList::clear() {
	for (unsigned int i = 0; i < nodes.size(); i++) if (nodes[i]) delete nodes[i];
	nodes.resize(0);
};

bool ListWdbeQTrgList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	vector<unsigned int> ics;
	vector<bool> shorttags;

	WdbeQTrgList* rec = NULL;

	string s;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "ListWdbeQTrgList");
	else
		basefound = checkXPath(docctx, basexpath);

	extractList(docctx, basexpath, "WdbeQTrgList", "row", "jnum", ics, shorttags);

	clear();

	for (unsigned int i = 0; i < ics.size(); i++) {
		rec = new WdbeQTrgList(ics[i]);

		if (shorttags[i]) s = basexpath + "/row[@jnum='" + to_string(ics[i]) + "']";
		else s = basexpath + "/WdbeQTrgList[@jnum='" + to_string(ics[i]) + "']";

		if (rec->readXML(docctx, s)) nodes.push_back(rec);
		else delete rec;
	};

	return basefound;
};

